doc/en/components/grids/_shared/conditional-cell-styling.md

```diff
@@ -0,0 +1,684 @@
+_keywords: conditional styling, {Platform}, {ProductName}, Infragistics
+mentionedTypes: [{ComponentApiMembers}]
+sharedComponents: ["Grid", "TreeGrid", "HierarchicalGrid"]
+---
+
+# {Platform} {ComponentTitle} Conditional Styling
+If you need to provide any custom styling in the {ComponentName} component, you can do it on either row or cell level.
+
+## {ComponentTitle} Conditional Row Styling
+
+The {ComponentName} component in Ignite UI for Angular provides two ways to **conditional styling of rows** based on custom rules.
+
+- By setting `RowClasses` input on the {ComponentName} component;
+- By setting `RowStyles` input on the {ComponentName} component;
+
+Further in this topic we will cover both of them in more details.
+
+### Using rowClasses
+You can conditionally style the {ComponentName} rows by setting the `RowClasses` input and define custom rules.
+
+
+```html
+<!-- sample.component.html -->
+<{ComponentSelector} #grid [data]="data" [height]="'600px'" [width]="'100%'" [rowClasses]="rowClasses">
+</{ComponentSelector}>
+```
+
+```razor
+ADD razor rowclasess
+```
+
+The `RowClasses` input accepts an object literal, containing key-value pairs, where the key is the name of the CSS class, while the value is either a callback function that returns a boolean, or boolean value.
+
+<!-- Angular -->
+```typescript
+// sample.component.ts
+
+public rowClasses = {
+    activeRow: this.activeRowCondition
+};
+
+public activeRowCondition = (row: RowType) => this.grid?.navigation.activeNode?.row === row.index;
+```
+
+```scss
+// sample.component.scss
+
+::ng-deep {
+ .activeRow {
+    border: 2px solid #fc81b8;
+    border-left: 3px solid #e41c77;
+ }
+}
+```
+<!-- end: Angular -->
+
+
+```razor
+Add snippet
+```
+
+<!-- Angular -->
+Use **::ng-deep** or **ViewEncapsulation.Non** to force the custom styles down through the current component and its children.
+<!-- end: Angular -->
+
+
+### Demo
+
+<!-- ComponentStart: Grid -->
+
+<code-view style="height:620px"
+           data-demos-base-url="{environment:demosBaseUrl}"
+           iframe-src="{environment:demosBaseUrl}/{ComponentSample}-rowClasses-sample"
+           github-src="{ComponentSample}/grid-rowClasses-sample" >
+</code-view>
+
+<!-- ComponentEnd: Grid -->
+
+
+<!-- ComponentStart: TreeGrid -->
+
+<code-view style="height:600px"
+           data-demos-base-url="{environment:demosBaseUrl}"
+           iframe-src="{environment:demosBaseUrl}/{ComponentSample}-classes"
+           github-src="{ComponentSample}/classes" >
+</code-view>
+
+<!-- ComponentEnd: TreeGrid -->
+
+<!-- ComponentStart: HierarchicalGrid -->
+
+<code-view style="height:600px"
+           data-demos-base-url="{environment:demosBaseUrl}"
+           iframe-src="{environment:demosBaseUrl}/{ComponentSample}-row-classes"
+           github-src="{ComponentSample}/row-classes" >
+           
+</code-view>
+
+<!-- ComponentEnd: HierarchicalGrid -->
```

Comment:
replace above 3 code-view blocks with just one code-view and remove component build flag because routing path to the sample will be the same for all grid samples in Sample Browser.